src/dswx_sar/metadata.py

```diff
@@ -0,0 +1,306 @@
+        if description:
+            dswx_metadata_dict[meta_key] = description
+        elif file_path:
+            dswx_metadata_dict[meta_key] = os.path.basename(file_path)
+        else:
+            dswx_metadata_dict[meta_key] = 'NOT_PROVIDED_OR_NOT_USED'
+
+
+def _populate_processing_metadata_datasets(dswx_metadata_dict, cfg):
+    """
+    Populate the metadata dictionary with processing information.
+    Parameters
+    ----------
+    dswx_metadata_dict: dict
+        Dictionary to be updated with processing metadata.
+    cfg: object
+        Configuration object containing processing metadata.
+    """
+    try:
+        processing_cfg = cfg.groups.processing
+        # Mapping for simple key-value assignments
+        threshold_mapping = {
+            'otsu': 'OTSU',
+            'ki': 'Kittler-Illingworth'
+        }
+        dswx_metadata_dict.update({
+            'POLARIZATION': processing_cfg.polarizations,
+            'FILTER': 'Enhanced Lee filter',
+            'THRESHOLDING': threshold_mapping.get(processing_cfg.initial_threshold.threshold_method),
+            'TILE_SELECTION': processing_cfg.initial_threshold.selection_method,
+            'MULTI_THRESHOLD': processing_cfg.initial_threshold.multi_threshold,
+            'FUZZY_SEED': processing_cfg.region_growing.initial_threshold,
+            'FUZZY_TOLERANCE': processing_cfg.region_growing.relaxed_threshold,
+            'INUNDATED_VEGETATION': processing_cfg.inundated_vegetation.enabled
+        })
+    except AttributeError as e:
+        print(f"Attribute error occurred: {e}")
+    except KeyError as e:
+        print(f"Key error occurred: {e}")
+    except Exception as e:
+        print(f"An unexpected error occurred: {e}")
+
+
+def compute_spatial_coverage(data_array):
+    """
+    Compute the spatial coverage.
+
+    Parameters
+    ----------
+    data_array : np.array
+        The 2D numpy array representation of the GeoTIFF.
+
+    Returns
+    -------
+    float
+        Spatial coverage as a percentage.
+    """
+    total_pixels = data_array.size
+    invalid_pixels = np.sum(data_array == 255)
+    valid_pixels = total_pixels - invalid_pixels
+
+    return round(valid_pixels / total_pixels * 100, 4)
+
+
+def compute_layover_shadow_coverage(data_array, spatial_coverage):
+    """
+    Compute the layover-shadow coverage.
+
+    Parameters
+    ----------
+    data_array : np.array
+        The 2D numpy array representation of the GeoTIFF.
+    spatial_coverage : float
+        Spatial coverage as a percentage.
+
+    Returns
+    -------
+    float
+        Layover-shadow coverage as a percentage.
+    """
+    layover_shadow_pixels = np.sum(data_array == band_assign_value_dict['layover_shadow_mask'])
+
+    if spatial_coverage > 0:
+        return round(layover_shadow_pixels / (spatial_coverage / 100 * data_array.size) * 100, 4)
+    else:
+        return np.nan
+
+
+def _populate_statics_metadata_datasets(dswx_metadata_dict, dswx_geotiff):
```

Comment:
Please check if this function is accessed from another code. Does the code in another PR use this function?

Reply:
This function is being used in line 253 of `save_mgrs_tiles.py`